src/types.ts

```diff
@@ -70,3 +70,47 @@ export interface UserSettings {
 export function isStreetViewImagery(x: string): x is "google" | "bing" {
   return x == "google" || x == "bing";
 }
+
+export type Mode =
+  | "edit-attribute"
+  | "edit-geometry"
+  | "route"
+  | "point"
+  | "free-polygon"
+  | "snap-polygon"
+  | "split-route"
+  | "street-view";
+
+export const MapEvents = [
```

Comment:
All of this new stuff could maybe stay scoped to `lib/draw`. It's specific to everything managed by Toolbox. I don't think we'll want to do the same event management elsewhere in ATIP, but kind of unknown till we explore contextual layer stuff more

Reply:
agreed, had this thought about the `eventHandler` types and `default_event_handler`